vmm/src/vstate.rs

```diff
@@ -253,17 +253,23 @@ impl Vcpu {
         kernel_start_addr: GuestAddress,
         vm: &Vm,
     ) -> Result<()> {
-        // the MachineConfiguration has defaults for ht_enabled and vcpu_count hence it is safe to unwrap
-        filter_cpuid(
+        let cpuid_vm_spec = VmSpec::new(
             self.id,
             machine_config
                 .vcpu_count
                 .ok_or(Error::VcpuCountNotInitialized)?,
             machine_config.ht_enabled.ok_or(Error::HTNotInitialized)?,
-            &mut self.cpuid,
         )
         .map_err(Error::CpuId)?;
 
+        let maybe_cpuid_main_transformer =
```

Comment:
I don't particularly like this part. We shouldn't care in `vstate` that `cpuid` has a transformer because this is an implementation detail of the `cpuid` crate.

I suggest keeping filter_cpuid function and move this code in that function. This is related to keeping the same abstraction as detailed here: https://en.wikipedia.org/wiki/Dependency_inversion_principle

Reply:
I was not challenging adding the transformer trait. I was just challenging exposing the fact we are using a transformer in cpuid. 

As you also state, the cpuid should expose two functionalities:
1. General filtering of cpuid entries (responsible for setting appropriate hypervisor flags or filtering other un-needed flags) and
2. Templating

These are the only functions that should leave the cpuid crate as the public interface.

Reply:
Ok. Done.